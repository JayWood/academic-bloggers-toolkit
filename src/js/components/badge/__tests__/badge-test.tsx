import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import * as React from 'react';

import Badge from '..';

describe('<Badge />', () => {
    test('defaults < 99', () => {
        let component = shallow(<Badge count={5} />);
        expect(toJSON(component)).toMatchSnapshot();

        component = shallow(<Badge count={50} />);
        expect(toJSON(component)).toMatchSnapshot();

        component = shallow(<Badge count={98} />);
        expect(toJSON(component)).toMatchSnapshot();
    });
    test('with color set < 99', () => {
        let component = shallow(<Badge count={5} color="magenta" />);
        expect(toJSON(component)).toMatchSnapshot();

        component = shallow(<Badge count={50} color="#ff00ff" />);
        expect(toJSON(component)).toMatchSnapshot();

        component = shallow(<Badge count={98} color="rgba(150, 150, 150, 0.9)" />);
        expect(toJSON(component)).toMatchSnapshot();
    });
    test('count >= 99', () => {
        let component = shallow(<Badge count={99} />);
        expect(toJSON(component)).toMatchSnapshot();

        component = shallow(<Badge count={175} color="magenta" />);
        expect(toJSON(component)).toMatchSnapshot();
    });
});
