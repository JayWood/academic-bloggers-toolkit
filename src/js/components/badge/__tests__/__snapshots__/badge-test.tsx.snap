// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> count >= 99 1`] = `
<div
  className="badge"
  style={
    Object {
      "background": undefined,
    }
  }
>
  99+
</div>
`;

exports[`<Badge /> count >= 99 2`] = `
<div
  className="badge"
  style={
    Object {
      "background": "magenta",
    }
  }
>
  99+
</div>
`;

exports[`<Badge /> defaults < 99 1`] = `
<div
  className="badge"
  style={
    Object {
      "background": undefined,
    }
  }
>
  5
</div>
`;

exports[`<Badge /> defaults < 99 2`] = `
<div
  className="badge"
  style={
    Object {
      "background": undefined,
    }
  }
>
  50
</div>
`;

exports[`<Badge /> defaults < 99 3`] = `
<div
  className="badge"
  style={
    Object {
      "background": undefined,
    }
  }
>
  98
</div>
`;

exports[`<Badge /> with color set < 99 1`] = `
<div
  className="badge"
  style={
    Object {
      "background": "magenta",
    }
  }
>
  5
</div>
`;

exports[`<Badge /> with color set < 99 2`] = `
<div
  className="badge"
  style={
    Object {
      "background": "#ff00ff",
    }
  }
>
  50
</div>
`;

exports[`<Badge /> with color set < 99 3`] = `
<div
  className="badge"
  style={
    Object {
      "background": "rgba(150, 150, 150, 0.9)",
    }
  }
>
  98
</div>
`;
